wandb/apis/public.py

```diff
@@ -4353,6 +4353,19 @@ def __init__(self, client, entity, project, name, attrs=None):
         self._attrs = attrs
         if self._attrs is None:
             self._load()
+
+        # The entity and project above are taken from the passed-in artifact version path
+        # so if the user is pulling an artifact version from an artifact portfolio, the entity/project
+        # of that portfolio may be different than the birth entity/project of the artifact version.
+        self._birth_project = (
+            self._attrs.get("artifactType", {}).get("project", {}).get("name")
+        )
+        self._birth_entity = (
+            self._attrs.get("artifactType", {})
+            .get("project", {})
+            .get("entity", {})
+            .get("name")
+        )
```

Comment:
This is a very JS-type pattern; I think a little more pythonic would be to use  `try`/`except`:
```suggestion
        self._birth_project, self._birth_entity = None, None
        try:
            project = self._attrs["artifactType"]["project"]
            self._birth_project = project["name"]
            self._birth_entity = project["entity"]["name"]
        except KeyError:
            pass
```
Here I'm assuming that if project.entity.name exists then project.name exists as well; otherwise you would need two try/except blocks.

Reply:
I don't prefer the two try/except blocks so I'll keep the current style unless you have a strong objection?